### DISCUSSION TITLE
Unexpected difference related to relative vs. absolute compiler executable paths

### DISCUSSION THREAD

**Discussion by tlemo** [compiler-explorer/compiler-explorer#3996] [General]
I ran into a small, but surprising, difference between using absolute or relative paths for compilers: the `initialise()` method is not called on the compiler object when relative paths are used. 

Here's the relevant code. Is this semantic difference intentional? (whether using relative paths for the compiler executable is a good idea or not is a different question)

https://github.com/compiler-explorer/compiler-explorer/blob/219f0610c4df1cc2c0b6e53c5f4737e5c6bd275f/lib/handlers/compile.js#L147-L172

Was the intention to set `compiler.exe = exe;` when resolving relative paths?

https://github.com/compiler-explorer/compiler-explorer/blob/219f0610c4df1cc2c0b6e53c5f4737e5c6bd275f/lib/handlers/compile.js#L145

**Comment by apmorton** [comment]
This is definitely a bug - compiler.exe should be set after resolving.

**Comment by tlemo** [comment]
Thanks @apmorton for confirming it. I converted this discussion to an issue (#3997) and created a tiny PR with the fix: #3998
